kubernetes/src/main/java/com/linecorp/armeria/client/kubernetes/ArmeriaHttpClientFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.kubernetes;
+
+import com.linecorp.armeria.client.WebClientBuilder;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.http.HttpClient;
+import io.fabric8.kubernetes.client.http.HttpClient.Factory;
+
+/**
+ * An {@link HttpClient.Factory} for creating {@link ArmeriaHttpClient}.
+ */
+public class ArmeriaHttpClientFactory implements HttpClient.Factory {
```

Comment:
`final` is not added by design to allow a subclass to override a client configuration via `.additionalConfig`.
https://github.com/fabric8io/kubernetes-client/blob/e4f434578f3c1343fb66660c53bc2de105466081/httpclient-okhttp/src/main/java/io/fabric8/kubernetes/client/okhttp/OkHttpClientFactory.java#L44-L49
By the way, it would be good to update Javadoc for `additionalConfig()`